tasks/gulp/__tests__/after-build-package.test.js

```diff
@@ -1,112 +1,83 @@
-        'package.json',
-        'govuk-prototype-kit.config.json',
-        '**/macro-options.json',
-        '**/fixtures.json',
-        'README.md'
-      ]
-
-      return recursive(configPaths.src, filesToIgnore).then(
-        files => {
-          let filesNotInSrc = files
-          // Use glob to generate an array of files that accounts for wildcards in filenames
-          filesNotInSrc = glob.sync('{' + additionalFilesNotInSrc.join(',') + '}', { cwd: 'package' })
-
-          return files
-            .map(file => {
-              // Remove /src prefix from filenames
-              const fileWithoutSrc = file.replace(/^src\//, '')
-
-              // Account for govuk-esm folder
-              if (fileWithoutSrc.split('.').pop() === 'mjs') {
-                const esmFile = fileWithoutSrc.replace('govuk/', 'govuk-esm/')
-                const umdFile = fileWithoutSrc.replace('.mjs', '.js')
-
-                return [umdFile, esmFile]
-              } else {
-                return fileWithoutSrc
-              }
-            })
-            // Allow for additional files that are not in src
-            .concat(filesNotInSrc)
-            .flat()
-            // Sort to make comparison easier
-            .sort()
-        },
-        error => {
-          console.error('Unable to get package files', error)
-        }
-      )
-    }
-
-    // Compare the expected directory listing with the files we expect
-    // to be present
-    return Promise.all([actualPackageFiles(), expectedPackageFiles()])
-      .then(results => {
-        const [actualPackageFiles, expectedPackageFiles] = results
-
-        expect(actualPackageFiles).toEqual(expectedPackageFiles)
-      })
+  it('should contain the expected files', async () => {
+    const filterPatterns = [
+      '!**/.DS_Store',
+      '!**/*.test.*',
+      '!**/__snapshots__/',
+      '!**/__snapshots__/**',
+      '!govuk/README.md'
+    ]
+
+    // Build array of expected output files
+    const listingExpected = listingSource
+      .filter(filterPath(filterPatterns))
+
+      // Replaces source component '*.mjs' with:
+      // - `*.mjs` file copied to `govuk-esm`
+      // - `*.js` file compiled to `govuk`
+      .flatMap(mapPathTo(['**/*.mjs'], ({ dir: requirePath, name }) => {
+        const importPath = normalize(slash(requirePath).replace(/^govuk/, 'govuk-esm'))
+
+        return [
+          join(importPath, `${name}.mjs`),
+          join(requirePath, `${name}.js`)
+        ]
+      }))
+
+      // Replaces source component '*.yaml' with:
+      // - `fixtures.json` fixtures for tests
+      // - `macro-options.json` component options
+      .flatMap(mapPathTo(['**/*.yaml'], ({ dir }) => [
+        join(dir, 'fixtures.json'),
+        join(dir, 'macro-options.json')
+      ]))
+
+      // Files already present in 'package'
+      .concat(...[
+        normalize('govuk-prototype-kit.config.json'),
+        normalize('package.json'),
+        normalize('README.md')
```

Comment:
*question*: Why are these strings normalised while there's no slashes in them?

Reply:
@romaricpascal Good spot, we use `cwd` in **glob()** to use relative paths so I'll remove these

(Previously the paths were `package/` suffixed)